In [1]:
import pandas as pd

In [2]:
titanic_df = pd.read_csv("train.csv")

In [3]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


# Specific Information

In [8]:
titanic_df[["Age","Fare"]]

Age     Fare
0    22.0   7.2500
1    38.0  71.2833
2    26.0   7.9250
3    35.0  53.1000
4    35.0   8.0500
..    ...      ...
886  27.0  13.0000
887  19.0  30.0000
888   NaN  23.4500
889  26.0  30.0000
890  32.0   7.7500

[891 rows x 2 columns]

In [9]:
titanic_df[["Age","Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [10]:
titanic_df[["Age","Fare"]].describe()

Age        Fare
count  714.000000  891.000000
mean    29.699118   32.204208
std     14.526497   49.693429
min      0.420000    0.000000
25%     20.125000    7.910400
50%     28.000000   14.454200
75%     38.000000   31.000000
max     80.000000  512.329200

### Summary of Age and Fare (Titanic Dataset)

The **Age** distribution is approximately normal, centered around **29 years**, with moderate spread. In contrast, the **Fare** distribution is **highly skewed**, showing significant outliers — with some passengers paying over **$500**, far above the average.


### Age and Fare Summary (Titanic Dataset)

- **Age:**
  - Average age is approximately **29.7 years**.
  - Middle 50% of passengers are aged between **20.1** and **38 years**.
  - Youngest passenger is **0.42 years**, and oldest is **80 years**.
  - Standard deviation is **14.5**, indicating moderate spread.
  - **714** non-null age entries (some missing values).

- **Fare:**
  - Average fare is approximately **$32.20**.
  - Middle 50% of fares range between **$7.91** and **$31.00**.
  - Minimum fare is **$0.00**; maximum is a high **$512.33**.
  - High **standard deviation of 49.7**, indicating wide variation.
  - Distribution is **highly skewed**, with a few passengers paying extremely high fares.


In [12]:
titanic_df.value_counts("Survived")

Survived
0    549
1    342
Name: count, dtype: int64

In [15]:
titanic_df["Sex"].value_counts()

Sex
male      577
female    314
Name: count, dtype: int64

In [16]:
titanic_df.value_counts(["Sex","Survived"])

Sex     Survived
male    0           468
female  1           233
male    1           109
female  0            81
Name: count, dtype: int64

In [21]:
titanic_df.value_counts(["Sex","Survived"], normalize = True, ascending = True)

Sex     Survived
female  0           0.090909
male    1           0.122334
female  1           0.261504
male    0           0.525253
Name: proportion, dtype: float64

Intrestingly, death rate of male population is double the rate of female who survived.

In [22]:
titanic_df[titanic_df["Sex"] == "female"].value_counts("Survived", normalize = True )

Survived
1    0.742038
0    0.257962
Name: proportion, dtype: float64

"The data shows that approximately one-third of the female population survived the accident, while around 25% did not."



In [40]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts("Survived")

Survived
0    427
1    217
Name: count, dtype: int64

In [38]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts("Survived", normalize = True)

Survived
0    0.663043
1    0.336957
Name: proportion, dtype: float64

In [48]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts(["Survived","Sex"])

Survived  Sex   
0         male      364
1         female    140
          male       77
0         female     63
Name: count, dtype: int64